src/foam/parse/parse.js

```diff
@@ -123,6 +123,10 @@ foam.CLASS({
                   'Cannot make substring: end PStream is not a tail of this.');
 
       return this.str[0].substring(this.pos, end.pos);
+    },
+
+    function apply(p, obj) {
```

Comment:
apply() is a method of functions in JS, so this might make the code (more) confusing for people because it will look the ps's are functions (I know, I'm the one who suggested the bad name). So a better name would be better.

Reply:
But can be done in a separate PR.